code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -865,3 +865,61 @@
 		log_game("DYNAMIC: Not enough players volunteered for the ruleset [name] - [candidates.len] out of [required_candidates].")
 		return FALSE
 	return TRUE
+
+//////////////////////////////////////////////
+//                                          //
+//           NINJA      (GHOST)             //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/from_ghosts/ninja
+	name = "Space Ninja"
+	midround_ruleset_style = MIDROUND_RULESET_STYLE_HEAVY
+	role_preference = /datum/role_preference/midround_ghost/ninja
+	required_type = /mob/dead/observer
+	antag_datum = /datum/antagonist/ninja
+	enemy_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_DETECTIVE, JOB_NAME_WARDEN, JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN)
+	required_enemies = list(2,2,2,2,2,2,2,2,2,2)
+	required_candidates = 1
+	weight = 3
+	cost = 9
+	minimum_players = 20
+	repeatable = TRUE
+	var/spawn_loc
+
+/datum/dynamic_ruleset/midround/from_ghosts/ninja/ready(forced)
+	if (!..())
+		return FALSE
+	//selecting a spawn_loc
+	var/list/spawn_locs = list()
+	for(var/obj/effect/landmark/carpspawn/L in GLOB.landmarks_list)
+		if(isturf(L.loc))
+			spawn_locs += L.loc
+	if(!length(spawn_locs))
+		log_game("DYNAMIC: [ruletype] ruleset [name] ready() failed due to no valid spawn locations (#1).")
+		return FALSE
+	spawn_loc = pick(spawn_locs)
+	if(!spawn_loc)
+		log_game("DYNAMIC: [ruletype] ruleset [name] ready() failed due to no valid spawn locations (#2).")
```

Comment:
there is no need for two checks here. if `length(spawn_locs)` is true, `pick()` will always be successful, and the loc cannot be null because you checked isturf.

Reply:
Will fix